dxm/src/main/java/io/pcp/parfait/dxm/PcpMmvWriter.java

```diff
@@ -307,9 +317,22 @@ public final <T> void registerType(Class<T> runtimeClass, TypeHandler<T> handler
      * @see io.pcp.parfait.pcp.PcpWriter#updateMetric(java.lang.String, java.lang.Object)
      */
     public final void updateMetric(MetricName name, Object value) {
-        if (!started) {
-            return;
+        // If another thread has called start() we need to wait until the writer has completely started before
+        // proceeding to update the metric value. This is to avoid a race condition where start() has already written
+        // the old metric value, but has not yet finished writing all of the values, when the metric is updated. The
+        // implementation here is a little complicated to avoid taking a lock on the happy paths.
+        if (state == State.STARTED) {
+            doUpdateMetric(name, value);
+        } else if (state == State.STARTING) {
+            if (stateMonitor.enterWhenUninterruptibly(isStarted, Duration.ofSeconds(10))) {
```

Comment:
I'm just channeling some things we observed during my Aconex days, where any of the synchronization overhead for metric updates caused grief. Particularly during startup with lots of metrics (this is why the QuiescentRegistryListener is there, just to help soften the blow of so many new metrics turning up during startup.)

Can we make the block wait time configurable rather than hard-coded? (leave the user to control how much they value this data loss over synchronization overhead?)

I presume with this tricky concurrency, no easy path to automate some unit tests here? (Man I hate doing that too, just asking...)

Reply:
@tallpsmith Would you like me to add the configuration to `MonitoringViewProperties`? Do you think 10 seconds is a reasonable default? Note that the duration is the maximum time it will wait. I would be surprised if it ever actually waited that long.

Reliable, cross-system concurrency testing is a can of worms. I can see if I can get anything to work tomorrow.

Reply:
@tallpsmith Just added a test that I think should be reliable. I confirmed it fails on the old code.

Reply:
@tallpsmith Just pushed another commit that makes the wait time configurable. Let me know if it's what you had in mind, and, if so, if you'd also like `parfait.sh` updated.

Reply:
I don't know much about `parfait.sh` (@natoscott  ?) - but I _think_ it's not necessary (or we can do it in another release if it's important enough).